staging/src/k8s.io/kube-scheduler/config/v1alpha1/types.go

```diff
@@ -198,9 +199,11 @@ type Plugins struct {
 type PluginSet struct {
 	// Enabled specifies plugins that should be enabled in addition to default plugins.
 	// These are called after default plugins and in the same order specified here.
+	// +listType=atomic
```

Comment:
It's confusing that Weight goes here and not in the config.

I'd consider separating the Enabled list, make it []string, leaving it atomic, and having a separate list of plugin params that is listType=map, and sets the Weight but doesn't determine the order. For your followups, not this PR.

Reply:
Uhm... we were not planning to change this. I'm not sure about the original rationale. @ahg-g @Huang-Wei ?

Reply:
Basically we can have these designs, the difference is quite trivial to me. (maybe @ahg-g knows more usecases)

- `Plugin` has Weight, `PluginConfig` holds Args (Args is optional and of type `runtime.Unknown`)
- Just one `PluginConfig` to hold Name, Weight, and Args.
- As @lavalamp said, `Plugin` to hold just name (hence can be refactored to []string in Enabled list), and `PluginConfig` to hold Name, Weight and Args.